### PR TITLE
Add toYYYYMMDDhhmmss function

### PR DESCRIPTION
**Pull request by ygf11**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

1. add toYYYYMMDDhhmmss and toYYYYMMDD function .
2. add doc.

## Changelog

- New Feature

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by sundy-li** [comment]
LGTM, let's add `toYYYYMMDD` into this pr?

**Comment by ygf11** [comment]
> LGTM, let's add toYYYYMMDD into this pr?

Yeah, that's ok.

File: common/functions/src/scalars/dates/number_function.rs [hunk 1]
```diff
@@ -26,44 +27,83 @@ use common_exception::Result;
...
         }))
     }
 }
 
-impl<T> Function for NumberFunction<T>
-where T: NumberResultFunction + Clone + Sync + Send + 'static
+impl<T, R> Function for NumberFunction<T, R>
+where
+    T: NumberResultFunction<R> + Clone + Sync + Send + 'static,
+    R: DFPrimitiveType + Clone + Sync + Send + 'static,
```
Comment (sundy-li) [inline common/functions/src/scalars/dates/number_function.rs:98]: DFPrimitiveType: NativeType

NativeType is already `Sync + Send + 'static`

File: common/functions/src/scalars/dates/number_function.rs [hunk 2]
```diff
@@ -80,15 +120,15 @@ where T: NumberResultFunction + Clone + Sync + Send + 'static
             DataType::Date16 => {
                 if let DataColumn::Constant(v, _) = columns[0].column() {
                     let date_time = Utc.timestamp(v.as_u64().unwrap() as i64 * 24 * 3600, 0_u32);
-                    let constant_result = Some(T::execute(date_time));
-                    Ok(DataColumn::Constant(DataValue::UInt32(constant_result), input_rows))
+                    let constant_result = T::to_constant_value(date_time);
+                    Ok(DataColumn::Constant(constant_result, input_rows))
                 }else {
```
Comment (sundy-li) [inline common/functions/src/scalars/dates/number_function.rs:125]: } else {

Comment (BohuTANG) [reply common/functions/src/scalars/dates/number_function.rs:125]: Hmm, why cargo CI can't warning this?

**Review by sundy-li — APPROVED** [review]
Looks great to me.

**Pull request merged**